scripts/run_notebooks_timeouts.py

```diff
@@ -0,0 +1,72 @@
+import os
+import subprocess
+import time
+
+from nbconvert.exporters import ScriptExporter
+
+TIMEOUT = 60  # seconds
+
+
+def convert_notebook_to_python(notebook_path, output_path):
+    # Add Matplotlib configuration to use a non-interactive backend
+    exporter = ScriptExporter()
+    content, _ = exporter.from_filename(notebook_path)
+    content = 'import matplotlib\nmatplotlib.use("Agg")\n' + content
+
+    with open(output_path, 'w', encoding='utf-8') as f:
+        f.write(content)
+
+
+def execute_python_script(script_path, timeout_seconds=30):
```

Comment:
Rather than convert to python files, would it be simpler to use the "execute" API of `nbconvert` directly?
https://nbconvert.readthedocs.io/en/latest/execute_api.html

I.e., 

```bash
jupyter nbconvert --to notebook --execute mynotebook.ipynb
```

Or perhaps it would be even simpler to use the python API, to avoid having to run in a subprocess? Eg:

```
with open(notebook_filename) as f:
    nb = nbformat.read(f, as_version=4)
ep = ExecutePreprocessor(timeout=TIMEOUT)
ep.preprocess(nb)
```

Reply:
Thanks, think this is a great solution